### ISSUE TITLE
[All] Error when same non-plus package is also present in project (directly or not) on Android

### ISSUE THREAD

**Issue by jasonlaw** [fluttercommunity/plus_plugins#49]
## System info

Issue occurs on: Android 
Plugin name: package_info_plus
Plugin version: 0.5.0
Flutter doctor output:

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 1.22.4, on Microsoft Windows [Version 10.0.17134.1845], locale en-US)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Android Studio (version 3.5)
[√] Connected device (1 available)

• No issues found!
```

## Steps to Reproduce

Hit error when run debugging.

## Logs

```
[        ] > Task :app:mergeDexDebug FAILED
[   +1 ms] C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info_plus\.transforms\559bdf8f2b898fb1f976337fe0e384aa\classes\classes.dex: D8:
Type io.flutter.plugins.packageinfo.BuildConfig is defined multiple times:
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info_plus\.transforms\559bdf8f2b898fb1f976337fe0e384aa\classes\classes.dex,
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info\.transforms\50b988ddcbec014edc04b574cf3502d9\classes\classes.dex
[   +3 ms] com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
[        ] Type io.flutter.plugins.packageinfo.BuildConfig is defined multiple times:
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info_plus\.transforms\559bdf8f2b898fb1f976337fe0e384aa\classes\classes.dex,
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info\.transforms\50b988ddcbec014edc04b574cf3502d9\classes\classes.dex
[        ] Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
[        ]      at com.android.builder.dexing.D8DexArchiveMerger.getExceptionToRethrow(D8DexArchiveMerger.java:132)
[        ]      at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:119)
[        ]      at com.android.build.gradle.internal.transforms.DexMergerTransformCallable.call(DexMergerTransformCallable.java:102)
[        ]      at com.android.build.gradle.internal.tasks.DexMergingTaskRunnable.run(DexMergingTask.kt:445)
[        ]      at com.android.build.gradle.internal.tasks.Workers$ActionFacade.run(Workers.kt:348)
[        ]      at org.gradle.workers.internal.AdapterWorkAction.execute(AdapterWorkAction.java:50)
[        ]      at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:47)
[        ]      at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1$1.create(NoIsolationWorkerFactory.java:65)
[        ]      at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1$1.create(NoIsolationWorkerFactory.java:61)
[        ]      at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:98)
[        ]      at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.execute(NoIsolationWorkerFactory.java:61)
[        ]      at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
[        ]      at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
[        ]      at
org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
[        ]      at
org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
[  +10 ms]      at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
[        ]      at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
[        ]      at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
[        ]      at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
[        ]      at org.gradle.workers.internal.NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:56)
[        ]      at org.gradle.workers.internal.DefaultWorkerExecutor$3.call(DefaultWorkerExecutor.java:215)
[        ]      at org.gradle.workers.internal.DefaultWorkerExecutor$3.call(DefaultWorkerExecutor.java:210)
[        ]      at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[        ]      at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:215)
[        ]      at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
[        ]      at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
[        ]      at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[        ]      at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[        ]      at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[        ]      at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
[        ]      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[        ]      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[        ]      at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
[        ]      at java.lang.Thread.run(Thread.java:748)
[        ] Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete
[        ]      at com.android.tools.r8.utils.O.a(:65)
[        ]      at com.android.tools.r8.D8.run(:11)
[        ]      at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:117)
[        ]      ... 34 more
[        ] Caused by: com.android.tools.r8.utils.b: Error:
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info_plus\.transforms\559bdf8f2b898fb1f976337fe0e384aa\classes\classes.dex, Type
io.flutter.plugins.packageinfo.BuildConfig is defined multiple times:
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info_plus\.transforms\559bdf8f2b898fb1f976337fe0e384aa\classes\classes.dex,
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info\.transforms\50b988ddcbec014edc04b574cf3502d9\classes\classes.dex
[   +1 ms]      at com.android.tools.r8.utils.y0.a(:21)
[        ]      at com.android.tools.r8.utils.x0.a(:30)
[        ]      at com.android.tools.r8.utils.x0.a(:10)
[        ]      at java.util.concurrent.ConcurrentHashMap.merge(ConcurrentHashMap.java:1990)
[   +1 ms]      at com.android.tools.r8.utils.x0.a(:6)
[        ]      at com.android.tools.r8.graph.A0$c.a(:6)
[        ]      at com.android.tools.r8.dex.a.a(:85)
[        ]      at com.android.tools.r8.dex.a.a(:12)
[        ]      at com.android.tools.r8.D8.d(:6)
[        ]      at com.android.tools.r8.D8.b(:1)
[        ]      at com.android.tools.r8.utils.O.a(:30)
[        ]      ... 36 more
[   +1 ms] FAILURE: Build failed with an exception.
[   +1 ms] * What went wrong:
[        ] Execution failed for task ':app:mergeDexDebug'.
[        ] > A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
[        ]    > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
[        ]      Type io.flutter.plugins.packageinfo.BuildConfig is defined multiple times:
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info_plus\.transforms\559bdf8f2b898fb1f976337fe0e384aa\classes\classes.dex,
C:\JSL\VIQCore\2020Flutter\viqcommunity_v3\build\package_info\.transforms\50b988ddcbec014edc04b574cf3502d9\classes\classes.dex
[   +1 ms]      Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full
insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 58s
[        ] 535 actionable tasks: 5 executed, 530 up-to-date
[ +364 ms] Picked up _JAVA_OPTIONS: -Xmx2g
[  +40 ms] Running Gradle task 'assembleDebug'... (completed in 59.7s)
[   +1 ms] [!] The shrinker may have failed to optimize the Java bytecode.
[        ]     To disable the shrinker, pass the `--no-shrink` flag to this command.
[        ]     To learn more, see: https://developer.android.com/studio/build/shrink-code
[   +3 ms] Exception: Gradle task assembleDebug failed with exit code 1
[   +1 ms] "flutter run" took 61,580ms.
[   +3 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:607:7)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:977:18)
           #3      _rootRunUnary (dart:async/zone.dart:1198:47)
           #4      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #5      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #7      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #8      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #11     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
           #12     _rootRunUnary (dart:async/zone.dart:1198:47)
           #13     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #14     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #15     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #16     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #17     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #18     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #19     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #20     AndroidDevice.isLocalEmulator (package:flutter_tools/src/android/android_device.dart)
           #21     _rootRunUnary (dart:async/zone.dart:1198:47)
           #22     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #23     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #24     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #25     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #26     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #27     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
           #28     _rootRun (dart:async/zone.dart:1190:13)
           #29     _CustomZone.run (dart:async/zone.dart:1093:19)
           #30     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #31     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #32     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #33     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #34     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #35     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
```

```
PS C:\JSL\VIQCore\2020Flutter\viqcommunity_v3> flutter analyze
Analyzing viqcommunity_v3...

   info - Unused import: 'package:flutter/foundation.dart' - lib\main.dart:4:8 - unused_import
   info - Unused import: 'dart:async' - lib\modules\auth\src\auth_controller.dart:1:8 - unused_import
   info - Unused import: 'package:flutter/foundation.dart' - lib\modules\auth\src\auth_controller.dart:2:8 - unused_import
   info - Unused import: 'package:flutter/gestures.dart' - lib\modules\auth\src\signup\signup_policy.dart:1:8 - unused_import
   info - Unused import: 'package:google_map_location_picker/google_map_location_picker.dart' - lib\modules\community\src\community_form.dart:3:8 -
          unused_import
   info - Unused import: 'package:google_maps_flutter/google_maps_flutter.dart' - lib\modules\community\src\community_form.dart:4:8 - unused_import
   info - Unused import: 'package:viqcommunity/modules/community/src/community_review_list.dart' -
          lib\modules\community\src\controllers\community_controller.dart:2:8 - unused_import

7 issues found. (ran in 16.2s)
```

```
[√] Flutter (Channel stable, 1.22.4, on Microsoft Windows [Version 10.0.17134.1845], locale en-US)
    • Flutter version 1.22.4 at C:\src\flutter
    • Framework revision 1aafb3a8b9 (10 days ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Android
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = C:\Android
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 42.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

• No issues found!
```

**Comment by balvinderz** [comment]
do you have both package_info and package_info_plus as dependencies in pubspec.yaml?

**Comment by jasonlaw** [comment]
Nope, I clean up explicitly the package_info before running the test, somehow it will generated out again.

**Comment by marcschw** [comment]
The problem is that package_info and package_info_plus share the same identifier and that many packages have package_info as a dependency. 
I just had the same error because I installed in_app_review which seems to be depending on  package_info but after removing in_app_review everything works fine. 

I'll have to use another package_info class though since I do need in_app_review.

**Comment by miquelbeltran** [comment]
Thanks for reporting this. Indeed this is a major breaking issue that happens with all packages.

It can be reproduced with these easy steps:

1. Select one package (e.g. android_intent_plus).
2. Open the example project.
3. Add the non-plus version of the package to the pubspec.yaml (e.g. android_intent)
4. Run the example.
5 See error in log:

```
D8: Program type already present: io.flutter.plugins.androidintent.BuildConfig
java.lang.RuntimeException: java.lang.RuntimeException: com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
Program type already present: io.flutter.plugins.androidintent.BuildConfig
```
The way to fix this will be to move all Android plugins to a different java package structure, e.g. `io.flutter.plugins.androidintentplus`.

**Comment by miquelbeltran** [comment]
Hi @jasonlaw, thanks for reporting the issue originally and sorry that it took us a while to fix it.

We have released version 0.6.0 of the package that fixes this issue by doing two things:

1. Move the Java files to another Java package to avoid the "already present" issue.
2. We renamed the method channel name, to also avoid collisions with an existing package_info.

Now both packages can be used at the same time without conflicts. I have tested this solution on Android and iOS.
